packages/common/src/getContract.ts

```diff
@@ -69,13 +72,17 @@ export function getContract<
   TAccount,
   TPublicClient,
   TWalletClient
->): GetContractReturnType<TAbi, TPublicClient, TWalletClient, TAddress> {
-  const contract = viem_getContract<TTransport, TAddress, TAbi, TChain, TAccount, TPublicClient, TWalletClient>({
+>): GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> {
+  const contract = viem_getContract({
     abi,
     address,
-    publicClient,
-    walletClient,
-  }) as unknown as GetContractReturnType<Abi, PublicClient, WalletClient>;
+    client: {
+      public: publicClient,
+      wallet: walletClient,
+    },
+  }) as unknown as GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> & {
+    write: unknown;
+  };
```

Comment:
The argument structure for viem's getContract have changed slightly, but mud's getContract still uses the same arguments to minimize the effect.

Reply:
since `getContract` is meant to align with/mirror viem, I wonder if we should update our arguments/types to math

Reply:
To ensure it remains a drop-in replacement, we should update the `mud_getContract` arguments to match those of viem v2. While mud users will need to update their code that uses `mud_getContract`, I now feel that this is natural.

OK, I'll go ahead and update the `mud_getContract` args.

```ts
// viem v2
getContract({
  abi,
  address,
  client: {
    public,
    wallet,
  },
})

// current mud
getContract({
  abi,
  address,
  publicClient,
  walletClient,
  onWrite?,
})

// will be...
getContract({
  abi,
  address,
  client: {
    public,
    wallet,
  },
  onWrite?,
})
```